active/0000-improved-actions.md

```diff
@@ -0,0 +1,278 @@
+    }
+  }
+});
+```
+
+```hbs
+{{! app/components/my-button/template.hbs }}
+{{my-button save=(action 'submit')}}
+```
+
+```js
+// app/components/my-button/component.js
+export default Ember.Component.extend({
+  click: function(){
+    this.attrs.save();
+    // for enhanced backwards compat, you may also this.sendAction('save');
+  }
+});
+```
+
+### Hole punching with a closure-based action
+
+The current system of action bubbleing falls down quickly when you want to pass a message through multiple
+levels of components. A closure based action system helps address this.
+
+Instead of relying on bubbling, a closure action wraps an action from the current context's
+`actions` hash in a function that will call it on that context. For example:
+
+```hbs
+{{! app/index/template.hbs }}
+{{my-form submit=(action 'submit')}}
+```
+
+```hbs
+{{! app/components/my-form/template.hbs }}
+{{my-button on-click=submit}}
+```
+
+```hbs
+{{! app/components/my-button/template.hbs }}
+{{my-button action=on-click}}
+```
+
+```js
+// app/components/my-button/component.js
+export default Ember.Component.extend({
+  click: function(){
+    this.attrs.action();
+    // for enhanced backwards compat, you may also this.sendAction();
+  }
+});
+```
+
+A closure action can also be called by an action handler:
+
+```hbs
+{{! app/index/template.hbs }}
+{{my-form submit=(action 'submit')}}
+```
+
+```hbs
+{{! app/components/my-form/template.hbs }}
+{{my-button on-click=submit}}
+```
+
+```hbs
+{{! app/components/my-button/template.hbs }}
+<button {{action on-click}}></button>
+```
+
+Lastly, closure actions allow for yielding an action to a block. For example:
+
+```hbs
+{{! app/index/template.hbs }}
+{{my-form save=(action 'submit') as |submit reset}}}
+  <button {{action submit}}>Save</button>
+  {{! ^ goes to my-form's save attr property, which
+        is the submit action on the outer scope }}
+  <button {{action reset}}>Reset</button>
+  {{! ^ goes to my-form }}
+  <button {{action "cancel"}}>Cancel</button>
+  {{! ^ goes to outer scope }}
+{{/my-form}}
+```
+
+```hbs
+{{! app/components/my-form/template.hbs }}
+{{yield save (action 'reset')}}
+```
+
+```js
+// app/components/my-form/component.js
+export default Ember.Component.extend({
+  actions: {
+    reset: function(){
+      // rollback
+    }
+  }
+});
+```
+
+# Drawbacks
+
+Currently `{{action` is only used in an element space:
+
+```hbs
+<button {{action "booyah"}}>Fire</button>
+```
+
+The closure usage is a new, perhaps `action` is not the right word. However the two
+behaviors are pretty similar in their conceptual behavior.
+
+* `{{action` in an element space attaches an action from the current scope to an event on the element,
+* `(action` closes over an action from the current scope so it can be attached as an action later.
+
+Additionally, there may be developers who still have `{{action someActionName}}` instead
+of the quoted version. This is long deprecated, but these apps may see some
```

Comment:
AFAIK, it is actually unsupported since 1.10 or 1.11.  We do support bound action names (where the name resolves to a string when clicked) though, and that is not deprecated (and I think it should stay).